js/utility/bc-google-analytics.js

```diff
@@ -79,13 +90,103 @@ bcpl.utility.googleAnalytics = (() => {
 		addOutboundLinkTracking();
 	};
 
+	const getDefaultKey = (action) => Object.prototype.hasOwnProperty.call(defaultGoogleEvents, action) ? 
+		defaultGoogleEvents[action] : 
+		null;
+
+	const getDefaultEvent = (event) => {
+		const {
+			action,
+			label
+		} = event;
+		const defaultKey = getDefaultKey(action);
+
+		return defaultKey ? 
+			{
+				[defaultKey]: label
+			} : null;
+	};
+
+	const getStandardEvent = (event) => {
+		const {
+			category,
+			label,
+			value
+		} = event;
+		const eventObj = {};
+
+		if (category) {
+			eventObj[googleKeys.category] = category;
+		}
+		if (label) {
+			eventObj[googleKeys.label] = label;
+		}
+		if (value) {
+			eventObj[googleKeys.value] = value;
+		}
+
+		return eventObj;
+	};
+
+	const getEventData = (event) => getDefaultEvent(event) || getStandardEvent(event);
+
+	const trackEvent = (event) => {
+		const eventData = getEventData(event);
+
+		gtag('event', event.action, eventData);
+	};
+
+	const trackLogin = (loginType) => {
+		const loginEvent = {
+			action: 'login',
+			label: loginType
+		};
+
+		trackEvent(loginEvent);
+	};
+
+	const trackSearch = (searchTerm) => {
+		const searchEvent = {
+			action: 'search',
+			label: searchTerm
+		};
+
+		trackEvent(searchEvent);
+	};
+
+	const trackShare = (shareType) => {
+		const shareEvent = {
+			action: 'share',
+			label: shareType
+		};
+
+		trackEvent(shareEvent);
+	};
+
+	const trackViewSearchResults = (searchTerm) => {
+		const viewSearchResultsEvent = {
+			action: 'view_search_results',
+			category: 'engagement',
```

Comment:
You shouldn't need category here, engagement will be used by default see https://developers.google.com/analytics/devguides/collection/gtagjs/events

Reply:
Good catch